compiler/noirc_frontend/src/ast/type_alias.rs

```diff
@@ -4,18 +4,89 @@ use noirc_errors::Location;
 use std::fmt::Display;
 
 /// Ast node for type aliases
+/// A Noir Type Alias can be an alias to a normal type, or to a numeric generic type.
 #[derive(Clone, Debug)]
-pub struct NoirTypeAlias {
+pub enum NoirTypeAlias {
+    NormalTypeAlias(NormalTypeAlias),
+    NumericTypeAlias(NumericTypeAlias),
+}
+
+impl Display for NoirTypeAlias {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NoirTypeAlias::NormalTypeAlias(alias) => write!(f, "{}", alias),
+            NoirTypeAlias::NumericTypeAlias(alias) => write!(f, "{}", alias),
+        }
+    }
+}
+#[derive(Clone, Debug)]
+pub struct NormalTypeAlias {
     pub name: Ident,
     pub generics: UnresolvedGenerics,
     pub typ: UnresolvedType,
     pub visibility: ItemVisibility,
     pub location: Location,
 }
 
-impl Display for NoirTypeAlias {
+impl Display for NormalTypeAlias {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let generics = vecmap(&self.generics, |generic| generic.to_string());
         write!(f, "type {}<{}> = {}", self.name, generics.join(", "), self.typ)
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct NumericTypeAlias {
+    pub type_alias: NormalTypeAlias,
+    pub numeric_type: UnresolvedType,
+}
+
+impl Display for NumericTypeAlias {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let generics = vecmap(&self.type_alias.generics, |generic| generic.to_string());
+        write!(
+            f,
+            "numeric ({}) type {}<{}> = {}",
```

Comment:
I don't know if we use this strings for compilation, but right now that output doesn't match the syntax that we use for parsing it.

Reply:
I guess it is now irrelevant after the switch to the option instead of NoirTypeAlias enum